src/tag_index.c

```diff
@@ -124,18 +123,12 @@ static int tokenizeTagString(const char *str, const FieldSpec *fs, char ***resAr
     if (tok) {
       // normalize the string
       if (!(flags & TagField_CaseSensitive)) { // check case sensitive
-        size_t origLen = strlen(tok);
-        size_t newLen = 0;
-        char *longer_dst = unicode_tolower(tok, origLen, &newLen);
+        char *longer_dst = unicode_tolower(tok, &toklen);
         if (longer_dst) {
           tok = longer_dst;
         } else {
           tok = rm_strndup(tok, MIN(toklen, MAX_TAG_LEN));
         }
-        if (origLen != newLen) {
-          toklen = newLen;
-          tok[newLen] = '\0';
```

Comment:
Why remove the null terminator?

Reply:
It is not needed.
if `(longer_dst)` this is a new allocated string returned by unicode_tolower() and it is null terminated,
otherwise, toklen was updated by `unicode_tolower()` and `rm_strndup()` returns a null terminated string.